pkg/apis/core/validation/validation.go

```diff
@@ -6490,6 +6490,7 @@ func validateTopologySpreadConstraints(constraints []core.TopologySpreadConstrai
 		if err := validateNodeInclusionPolicy(subFldPath.Child("nodeTaintsPolicy"), constraint.NodeTaintsPolicy); err != nil {
 			allErrs = append(allErrs, err)
 		}
+		allErrs = append(allErrs, validateMatchLabelKeys(subFldPath.Child("matchLabelKeys"), constraint.MatchLabelKeys, constraint.LabelSelector)...)
 	}
```

Comment:
pre-existing, but I noticed validateTopologySpreadConstraints doesn't seem to validate constraint.LabelSelector at all... @ahg-g, can you open a separate issue to track that (need to make sure the scheduler is robust against completely invalid selectors, and consider how to fix this validation in the least disruptive way possible)

Reply:
Some update on this comments, the validation is actually done when the `labels.Selector` is built. e.g. 
https://github.com/kubernetes/kubernetes/blob/891cbede96ab4b64c48edf25f8bbd331d8731622/pkg/scheduler/framework/plugins/podtopologyspread/common.go#L94

And it is validated here, 
https://github.com/kubernetes/kubernetes/blob/891cbede96ab4b64c48edf25f8bbd331d8731622/staging/src/k8s.io/apimachinery/pkg/labels/selector.go#L191-L194

I think the original comment is still valid and the the validation mentioned above would mitigate the problem as the invalid label value won't be accepted at all.

Reply:
that happens at time of use, not time of API write, which is not ideal (since the user has no idea the object they created is not working like they want it to)

Reply:
pr sent: https://github.com/kubernetes/kubernetes/pull/112121